update.sh

```diff
@@ -45,16 +45,26 @@ generated_warning() {
 	EOH
 }
 
+# Equivalent to PHP's PHP_VERSION_ID:
+# 5.6.10 => 50610, 7.1.0 => 70100
+as_version_id() {
+	local fullVersion="$1"
+	# strip prelease suffix
+	fullVersion="${fullVersion/[a-z]*/}"
+	local components=(${fullVersion//./ })
+	if [ "${#components[@]}" -lt 2 ]; then
+		components+=(0)
+	fi
+	if [ "${#components[@]}" -lt 3 ]; then
+		components+=(0)
+	fi
+	printf '%d%02d%02d' ${components[@]}
```

Comment:
I'd prefer a tiny bit more whitespace and quoting here, but I also think we could probably simplify this by being a bit more explicit about what we pass to `printf`:

```suggestion
	printf '%d%02d%02d' "${components[0]}" "${components[1]:-0}" "${components[2]:-0}"
```